src/Temporalio.Extensions.Hosting/TemporalWorkerService.cs

```diff
@@ -113,27 +114,72 @@ public TemporalWorkerService(
         /// WARNING: Do not rely on the signature of this constructor, it is for DI container use
         /// only and may change in incompatible ways.
         /// </remarks>
-        [ActivatorUtilitiesConstructor]
         public TemporalWorkerService(
             (string TaskQueue, string? BuildId) taskQueueAndBuildId,
             IOptionsMonitor<TemporalWorkerServiceOptions> optionsMonitor,
             ITemporalClient? existingClient = null,
             ILoggerFactory? loggerFactory = null)
+        : this(
```

Comment:
If the `BuildId` worker option is deprecated, this constructor should be too IMO